db/manager.go

```diff
@@ -115,9 +118,13 @@ func (m *Manager) GetToken(ctx context.Context, identity thread.Identity) (threa
 
 // NewDB creates a new db and prefixes its datastore with base key.
 func (m *Manager) NewDB(ctx context.Context, id thread.ID, opts ...NewManagedOption) (*DB, error) {
```

Comment:
Going the route of tight locking instead of `defer` since the middle-body logic might be long-running and that would be too much locking.